apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -121,7 +124,15 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
         /// </summary>
```

Comment:
Don't we also want to change the AddScoped calls a few lines above to AddSingleton? Basically everything becomes a singleton, via the magic of IHttpContextAccessor?